transaction/depot.go

```diff
@@ -117,6 +126,9 @@ func NewDepot(handler DeliveryCourier, storage DepotStorage, nonce *DepotNonceTr
 func (d *Depot) Run() {
 	for _, s := range d.config.Workers {
 		go d.watchDeliveries(s)
+		if s.CleanupDaysLimit > 0 && s.CleanupInterval > 0 && s.CleanupLimit > 0 {
+			go d.cleanupOldEntries(s)
```

Comment:
Do you really need a separate cleaner goroutine for each worker? I think it's really enough to just clean up after X in a single routine. Simply filter by timestamp.